forch/faucet_state_collector.py

```diff
@@ -569,24 +623,35 @@ def _fill_path_to_root(self, switch_name, switch_map):
         """populate path to root for switch_state"""
         switch_map["root_path"] = self.get_switch_egress_path(switch_name)
 
-    def _fill_dva_states(self, switch_name, port_id, port_map):
-        dp_obj = self.faucet_config.get(DPS_CFG, {}).get(switch_name)
-        if not dp_obj:
+    def _fill_dva_states(self, switch_name, port_id, port_map, list_acls=False):
+        dp_config = self.faucet_config.get(DPS_CFG, {}).get(switch_name)
+        if not dp_config:
             LOGGER.warning('Switch not defined in dps config: %s', switch_name)
             return
 
-        port_obj = dp_obj.interfaces.get(port_id)
-        if not port_obj:
+        port_config = dp_config.ports.get(port_id)
+        if not port_config:
             LOGGER.warning('Port not defined in dps config: %s, %s', switch_name, port_id)
             return
 
-        native_vlan = port_obj.get('native_vlan')
-        acls_in = port_obj.get('acls_in')
-
-        if native_vlan:
-            port_map['vlan'] = int(native_vlan)
-        if acls_in:
-            port_map['acls'] = list(acls_in)
+        if port_config.native_vlan:
+            port_map['vlan'] = int(port_config.native_vlan.vid)
+        if port_config.acls_in:
+            if list_acls:
+                port_map['acls'] = [acl._id for acl in port_config.acls_in]
+            else:
+                acls_state = port_map.setdefault('acls', {})
+                for acl in port_config.acls_in:
+                    acl_state = acls_state.setdefault(str(acl._id), {})
+                    rules_state = acl_state.setdefault('rules', {})
+                    for rule in acl.rules:
+                        cookie = str(rule.get('cookie'))
+                        if not cookie:
+                            LOGGER.warning(
+                                'Rule in ACL %s does not have a cookie: %s', acl._id, rule)
+                            continue
```

Comment:
why does that make the rest invalid? (the continue)

Reply:
removed continue